src/main/kotlin/week1/racingcar/RacingGame.kt

```diff
@@ -1,36 +1,62 @@
 package week1.racingcar
 
-class RacingGame {
+import kotlin.random.Random
 
-    private var numberOfCars = 0
-    private var timesToRace = 0
+class RacingGame {
 
-    fun start() {
-        InputView().startInput().also {
-            this@RacingGame.apply {
-                numberOfCars = it.numberOfCars
-                timesToRace = it.timesToRace
-            }
-            race()
+    fun getInputResult(): InputResult {
+        return InputView().let {
+            it.getValidateRaceInfo(it.startInput())
         }
     }
 
-    private fun race() {
+    fun startGame(inputResult: InputResult) {
+        val cars = inputResult.cars.map { Car(it) }
+        val timesToRace = inputResult.timesToRace
+        race(cars, timesToRace)
+    }
+
+    private fun race(cars: List<Car>, timesToRace: Int) {
         val resultView = ResultView()
-        val cars = List(numberOfCars) { Car() }
-        (0 until timesToRace).forEach { _ ->
-            cars.forEach {
-                if (isPossibleToDrive()) {
-                    it.go()
-                }
-            }
-            resultView.showResult(cars)
+        repeat(timesToRace) {
+            drive(cars)
+            showResult(resultView, cars)
+        }
+    }
+
+    private fun drive(cars: List<Car>) {
+        cars.forEach {
+            if (isPossibleToDrive()) it.go()
+        }
+    }
+
+    private fun showResult(resultView: ResultView, cars: List<Car>) {
+        resultView.run {
+            showResult(cars)
+            showWinner(getWinners(cars))
         }
     }
 
-    private fun isPossibleToDrive() = RandomUtils.getRandomDigit() > 4
+    private fun isPossibleToDrive() = Random.nextInt(COUNT_OF_DIGITS) >= MIN_NUM_TO_GO
+
+    private fun getWinners(cars: List<Car>): List<String> {
+        val farthestMileage = getFarthestMileage(cars)
+        return cars.filter { it.mileage == farthestMileage }.map { it.name }
```

Comment:
가급적 하나의 하나의 `.`를 사용하도록 해보세요! 😃
가독성도 좋아지고 또 chain이 3개이상 연견되면 인텔리제이가 반환타입을 각 라인에 명시해줘서 코드를 파악하기 쉽게 해줍니다. 🤗

```suggestion
        return cars.filter { it.mileage == farthestMileage }
                .map { it.name }
```

<img width="559" alt="image" src="https://github.com/next-step/kotlin-racingcar/assets/47850258/06166c4f-e194-4d48-a78b-00526cf65053">